src/nvim/terminal.c

```diff
@@ -320,7 +320,7 @@ void terminal_open(Terminal **termpp, buf_T *buf, TerminalOptions opts)
   term->vts = vterm_obtain_screen(term->vt);
   vterm_screen_enable_altscreen(term->vts, true);
   // TODO(clason): reenable when https://github.com/neovim/neovim/issues/23762 is fixed
```

Comment:
@zeertzjq I've pulled in the fixes suggested in https://github.com/neovim/neovim/issues/23762 and haven't seen any issues so far. I'd appreciate it if you could test your reproducers and see if they still fail.

Reply:
I can't reproduce the heap-buffer-overflow on this branch.
```suggestion
```